src/pages/index.js

```diff
@@ -70,31 +141,95 @@ function handleImageClick(name, link) {
   newImagePopup.open({ link, name });
 }
 
-const cardSection = new Section(
-  {
-    items: cardData,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.getView();
-  cardListEl.prepend(cardElement);
+  cardSection.addItem(cardElement);
 }
 
-cardSection.renderItems();
+function handleDeleteClick(card) {
+  newPopupConfirm.open();
+  newPopupConfirm.setSubmitAction(() => {
+    api.deleteCard(card.id).then(() => {
+      card.deleteCard();
+      newPopupConfirm.close();
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**

Reply:
Okay I believe I got them all